macro/src/derive.rs

```diff
@@ -0,0 +1,361 @@
+use std::collections::VecDeque;
+
+use proc_macro2::Span;
+use quote::quote_spanned;
+use rust_format::Formatter;
+use syn::{spanned::Spanned, Attribute};
+use synstructure::VariantInfo;
+
+use crate::{argument::MethodSelector, parse::Parser, reflect::Reflector, signature::Signature};
+
+pub fn derive_to_rust(s: synstructure::Structure) -> proc_macro2::TokenStream {
+    let mut driver = Driver {
+        input: &s,
+        reflector: &mut Reflector::default(),
+    };
+    match driver.try_derive_to_rust() {
+        Ok(t) => {
+            if let Ok(debug_env) = std::env::var("DUCHESS_DEBUG") {
```

Comment:
[aside] is there a way in the macro system to have it dump the generated code with a standard debug/verbose option? Seems like a we have a lot of copies of this in places